### ISSUE TITLE
No warning issued for using a variable defined inside if statement

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#12]
Originally reported by: **Andrew Carman (BitBucket: [carmandrew](http://bitbucket.org/carmandrew), GitHub: @carmandrew?)**

----------------------------------------

Pylint should issue a warning when you use a variable that's defined inside an if statement. 
```
#!python

if False:
    foo = 7
print foo
```
This has no pylint errors, but raises an exception when run.

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/12

**Comment by PCManticore** [comment]
Closing in favor of #85 (merging the same feature request)
